### ISSUE TITLE
ANTS Build errors

### ISSUE THREAD

**Issue by mwagshul** [ANTsX/ANTs#1212]
Running into problems with build, ANTs downloaded through the installANTs.sh script. Files attached below.

Thanks for any advice/assistance with getting this fixed.

Mark

Before opening an issue, please review the build documentation here

https://github.com/ANTsX/ANTs/wiki/Compiling-ANTs-on-Linux-and-Mac-OS

If you are building the latest code snapshot, please also check Travis
to see if the code is building successfully

https://travis-ci.org/ANTsX/ANTs

If you are using system ITK or VTK, please verify you have the correct
versions (see the Wiki link above). If you can, please also attempt a
Superbuild and let us know if that compiles successfully.

**When did the error occur?**
[X ] Compilation (make)

**Build environment**
 - OS: [e.g. Linux, CentOS]
 - OS version: [6.9]
 - Type of system: [HPC cluster]
 
   
**ANTs version**
Latest  version

**Build configuration and logs**
Please attach the following files (relative to your build directory)

[build.log](https://github.com/ANTsX/ANTs/files/6875047/build.log)
[cmake.log](https://github.com/ANTsX/ANTs/files/6875048/cmake.log)
[CMakeCache.txt](https://github.com/ANTsX/ANTs/files/6875049/CMakeCache.txt)
[CMakeError.log](https://github.com/ANTsX/ANTs/files/6875050/CMakeError.log)
  - build.log (the terminal output from make)
  - CMakeCache.txt
  - CMakeFiles/CMakeError.log
  - CMakeFiles/CMakeOutput.log

**Additional context**

**Comment by cookpa** [comment]
It's failing to even start building ITK. Looks like it's finding the C / C++ compilers inside your conda environment, but it's not able to build a simple executable. I'm not sure what's going on.

What is your CMake version?

**Comment by mwagshul** [comment]
3.21.0.

Mark

**Comment by cookpa** [comment]
OK so your C++ and CMake version are up to date, which is good. 

I'm just guessing now but maybe changing the optimization flags by adding

```
-DSuperBuild_ANTS_C_OPTIMIZATION_FLAGS="-mtune=native -march=native" \
-DSuperBuild_ANTS_CXX_OPTIMIZATION_FLAGS="-mtune=native -march=native"
```

to the cmake command could help.

**Comment by mwagshul** [comment]
Hi. The problem seems to have been with my conda environment, although I
can't use my system's cmake since it's an old version, so I'm still using
the cmake within the conda folder, just not entering the conda environment.

But there are a few things I had to do to get this to work, and I am now
getting some errors which I wasn't getting before.

1) When I attempted to run cmake, I got an error with
"CXX_HAS_DISABLE_OPTIMIZATION_FLAG" - the cmake fails at this point, not
sure whether I can proceed with this failing.
I attempted to get around this by editing the cmake file (yes, I know, not
recommended), by commenting out the code related to this option, but I'm
not sure if this will have any untold downstream effects.
I noticed a comment online from a while back with the recommendation to set
the CMAKE_BUILD_TYPE to a null string, but that didn't help.
 2) Next, running the make file, I run into a number of errors:

**Comment by mwagshul** [comment]
Sorry, I realized that I had hit Send midstream in my last message!

I run into the following errors when running the make file:
1) The same error with the CXX_HAS_DISABLE_OPTIMIZATION_FLAG
2) Problems with the GIT_TAG's for a number of the github calls.

Again, I can "fix" this by commenting out the
CXX_HAS_DISABLE_OPTAIMIZATION_FLAG in the build/CMAKE folder, and removing
the GIT_TAG's (edits done midstream, after these files are created, but
prior to their being accessed - I'm sure there is a more "legal" way to do
this, by I'm reaching for straws right now!).

In any event, all of this leads to the compilation running much further,
but gets stuck much later downstream in not finding certain elements in the
ANTs/Examples folder.

Any further thoughts would be most appreciated!

Best,

Mark

**Comment by cookpa** [comment]
I did notice you had a very old git, I think it was 1.7 or something. That might be the source of those problems.

I used to build on a CentOS 6 cluster but it was recently decommissioned so I can't get the details. I think it was using GCC 4.9 though.

**Comment by mwagshul** [comment]
I think that fixed the problem (installed git 2.23 into my conda
environment), except that now I run into the original issue I described
(having to do with the BigEndian test).

To get around that problem, I was running cmake outside of conda, but then
the git clone for ITK defaults to the system git (ver 1.7). I tried setting
an alias for git to point to the new installed git version, but within the
build it goes back to the system git.

Is there a way for me to specify the path to git in the build? That would
solve the problem (without having to go through my IT to get them to update
git, which will take a long time).

Thanks,

Mark
